### ISSUE TITLE
Different results between Keras and TFJS - TFJS different answers on refresh

### ISSUE THREAD

**Issue by neerajmorar** [tensorflow/tfjs#1138]
To get help from the community, we encourage using Stack Overflow and the [`tesnorflow.js`](https://stackoverflow.com/questions/tagged/tensorflow.js) tag.

#### TensorFlow.js version

0.14.1 delivered via Script Tag 
`<script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@0.14.1/dist/tf.min.js"></script>`

#### Browser version

Google Chrome 71.0.3578.98 (64-bit)

#### Describe the problem or feature request

I have a model that tries to distinguish breeds of dogs consisting of 7 classes. This has been done via Python using Keras. However, after converting the model to JSON format and deploying it TFJS I do not get the same answers. Moreover, upon page refresh the model changes its answers for the same image, whereas in Keras the answer is always the same.

However, there may be a possibility that this is happening with the fact that I used `strict=false` in `tf.loadModel()` because it was giving me the error `Error: Provided weight data has no target variable: conv1_1_15/kernel` and setting that flag stopped it from happening.

I've used a different model before and that worked fine, the model design was exactly the same except instead of 7 classes it was 2 and it instead classified between dogs and cats. However, for this model `strict=false` wasn't needed, hence why I believe this may be the cause of this issue.

#### Code to reproduce the bug / link to feature request

Here's my model design - nothing spectacular:

![image](https://user-images.githubusercontent.com/12702488/51689186-12d2e780-1fee-11e9-9710-f7e2cdcb7498.png)

As you can see it, it's just a couple sections of CovNets followed by pooling and dropout, followed by a Flatten and Dense layer.

After saving the model using `model.save("tensorflowJS/model.h5")` I converted it using the command `tensorflowjs_converter --input_format keras tensorflowJS/model.h5 tensorflowJS` and got the expected JSON file and weights sharding.

Deploying the model for TFJS I created the following JavaScript code:

```
<script>
    var model;

    async function loadModel() {
        model = await tf.loadModel("model.json", strict=false);

        prediction();
    }

    async function prediction() {
        dog_breed = ["Beagle", "Neither", "German Shepherd", "Labrador", "Golden Retriever", "Border Collie", "Cocker Spaniel"];

        src = document.getElementById("image");
        src = src.src;

        var img = new Image();
        img.src = src;
        img.alt = "alt";
        img.width = 100;
        img.height = 100;

        var tensor = tf.fromPixels(img);

        const shape = [1, 100, 100, 3];
        tensor = tensor.reshape(shape);

        const pred = model.predict(tensor);
        maxValue = tf.argMax(pred, 1);
        index = maxValue.get(0);

        document.getElementById("label").innerHTML = dog_breed[index];
    }

    loadModel();
</script>
```

**Comment by neerajmorar** [comment]
Okay, I've fixed the issue and it was definitely to do with the `strict=false` flag. I re-saved my model in Keras after restarting the Kernel and re-converted it to JSON and this time it was able to load correctly without the need for that flag - I'm now getting the right answers.
